addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,339 @@
+    if (!shouldRunOnThisPage) {
+      return;
+    }
+
+    await gatherTopics();
+    // console.log("Topic Count: " + topics.length);
+    let highestTopicOnThisPage = highestTopicIdFrom(topics);
+    // console.log("Highest TopicId: " + highestTopicOnThisPage);
+    let lowestNewTopicId = highestTopicOnThisPage - determineTopicInterval();
+    highlightTopicCellsWithTopicIdBelow(topics, lowestNewTopicId);
+  }
+
+  /**
+   * Populates the topics array with an entry for each cell associated with a forum being handled.
+   * In most cases that means all topic cells, but Search pages are special.
+   * Any topics gathered previously will be forgotten. (And should have already been when disabling.)
+   * Each topic is:
+   *    - topicId      int, used to compare topic "age" relative to other posts on the page
+   *    - topicCell    Node that will be highlighted if too old
+   *    - forum        String, usually the forum being listed. Comes from the row on a search page
+   *    - restoreCell  Only defined if the cell is modified.
+   *                   A clone of the initial state of the cell Node, to revert for disable addon
+   */
+  async function gatherTopics() {
+    topics = [];
+    let possibleTopicCells; // verified before forming a topic
+    if (isMobileSite) {
+      possibleTopicCells = document.querySelectorAll(".topic.list>li");
+    } else {
+      possibleTopicCells = document.querySelectorAll(".tcl");
+    }
+    // console.log("possibleTopicCells length: " + possibleTopicCells.length);
+    let theForum = forumOrSearchPageName; // usually, but overridden per cell if on a search page
+    // console.log("theForum: " + theForum);
+    for (let i = 0; i < possibleTopicCells.length; i++) {
+      if (forumOrSearchPageName.includes(searchResultsPageName)) {
+        // the cell to the right of the topic cell lists the forum, in search results
+        if (!possibleTopicCells[i].nextElementSibling) {
+          // synthetic "Page n" rows attached by Infinite Scrolling on search pages
+          // have no sibling, because they span all 4 columns
+          continue;
+        }
+        theForum = possibleTopicCells[i].nextElementSibling.innerText;
+      }
+      const theTopicId = await extractTopicIdFrom(possibleTopicCells[i]);
+      if (theTopicId != 0) {
+        // No restoreCell property at this point. Only added if the cell is modified
+        const topic = {
+          topicId: theTopicId,
+          topicCell: possibleTopicCells[i],
+          forum: theForum,
+        };
+        topics.push(topic);
+        // console.log("gatherTopics: " + topic.topicId + " in " + topic.forum);
+      }
+    }
+  }
+
+  /**
+   * How many topics since the earliest on the same page to be considered a "necropost"
+   */
+  function determineTopicInterval() {
+    if (!addon.settings.get("chooseCustomTopicInterval")) {
+      return defaultTopicInterval;
+    }
+    return addon.settings.get("customTopicInterval");
+  }
```

Comment:
You should just return the `customTopicInterval` setting, since it's not nullish?

Reply:
chooseXXX is a boolean switch in the configuration. If the user has the switch off (false) we use the defaultTopicInterval. If chooseXXX is true, we use the value supplied by the user in customTopicInterval.

Reply:
Why have a switch though? We could just use the bare setting (keep in mind people can reset back to default settings easily)

Reply:
I didn't want to put the strange detail of how age is determined in front of the 99% of users who won't care. But it'll probably look fine. I'll try it out, see how it feels, and probably commit your suggestion.